packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -2254,6 +2254,19 @@ export class Bucket extends BucketBase {
         throw new Error('ExpiredObjectDeleteMarker cannot be specified with expiration, ExpirationDate, or TagFilters.');
       }
 
+      if (
```

Comment:
Thanks for the PR @go-to-k  and the unit testing - I'm now wondering how we missed this validation in the first place.

This is certainly a big `if` statement here, but I suppose it has to be for this validation. Trying to think of other ways to format this ...

```ts
const ruleProperties = [
  rule.abortIncompleteMultipartUploadAfter,
  rule.expiration,
  rule.expirationDate,
  rule.expiredObjectDeleteMarker,
  rule.noncurrentVersionExpiration,
  rule.noncurrentVersionsToRetain,
  rule.noncurrentVersionTransitions,
  rule.transitions,
];

if (ruleProperties.every(prop => prop === undefined)) {
  throw new Error('All rules for `lifecycleRules` must have at least one of the following properties: `abortIncompleteMultipartUploadAfter`, `expiration`, `expirationDate`, `expiredObjectDeleteMarker`, `noncurrentVersionExpiration`, `noncurrentVersionsToRetain`, `noncurrentVersionTransitions`, or `transitions`');
}
```

Approving this PR 👍